src/requests/sign-request.ts

```diff
@@ -113,11 +117,22 @@ export const signRequest = (
   const headers = canonicalRequest.headers ? normalizeHeaders(canonicalRequest.headers) : {}
   const body = canonicalRequest.body ?? ''
 
-  const { sortedHeaders, signedHeaders } = getSortedAndSignedHeaders(headers, timestamp)
+  const { sortedHeaders, signedHeaders } = getSortedAndSignedHeaders(
+    { ...headers, ...((contextHeaders as unknown) as Record<string, string>) },
+    timestamp
+  )
+  const subject = contextHeaders[ContentfulUserIdHeader]
+    ? { [ContentfulUserIdHeader]: contextHeaders[ContentfulUserIdHeader] }
+    : contextHeaders[ContentfulAppIdHeader]
+    ? { [ContentfulAppIdHeader]: contextHeaders[ContentfulAppIdHeader] }
+    : {}
```

Comment:
nit: a gool ol' if-else wrapped in a function would be much easier to read for me